crates/swc_ecma_transforms_module/src/system_js.rs

```diff
@@ -0,0 +1,1143 @@
+use crate::{
+    path::{ImportResolver, NoopImportResolver},
+    util::{local_name_for_src, use_strict},
+};
+use anyhow::Context;
+use serde::{Deserialize, Serialize};
+use swc_atoms::{js_word, JsWord};
+use swc_common::{collections::AHashMap, FileName, Mark, Span, DUMMY_SP};
+use swc_ecma_ast::*;
+use swc_ecma_utils::{
+    ident::IdentLike, member_expr, private_ident, quote_ident, quote_str, undefined,
+    var::VarCollector, ExprFactory,
+};
+use swc_ecma_visit::{noop_fold_type, Fold, FoldWith, VisitWith};
+#[derive(Debug, Clone, Default, Serialize, Deserialize)]
+#[serde(deny_unknown_fields, rename_all = "camelCase")]
+pub struct Config {
+    #[serde(default)]
+    pub allow_top_level_this: bool,
+}
+
+struct SystemJs<R>
+where
+    R: ImportResolver,
+{
+    root_mark: Mark,
+    resolver: Option<(R, FileName)>,
+    config: Config,
+
+    declare_var_idents: Vec<Ident>,
+    export_map: AHashMap<Id, Vec<JsWord>>,
+    export_names: Vec<JsWord>,
+    export_values: Vec<Expr>,
```

Comment:
Please use `Vec<Box<Expr>>` instead.

Reply:
It will be faster because you have to box it anyway, and you can prevent `memmove` of a large type (`Expr`) by doing so.

Reply:
If clippy complains, uou can add `#[allow(clippy::vec_box)]` or `#![allow(clippy::vec_box)]` on `lib.rs`